### PR TITLE
Acrolinx 1/2

### PR DESCRIPTION
**Pull request by gewarren**

### CODE REVIEW

File: docs/csharp/getting-started/introduction-to-the-csharp-language-and-the-net-framework.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ C# is an elegant and type-safe object-oriented language that enables developers
   
 ## C# Language
```
Comment (Youssef1313) [inline docs/csharp/getting-started/introduction-to-the-csharp-language-and-the-net-framework.md]:
```suggestion
## C# language
```

File: docs/csharp/getting-started/introduction-to-the-csharp-language-and-the-net-framework.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ C# is an elegant and type-safe object-oriented language that enables developers
...
+ C# syntax is highly expressive, yet it is also simple and easy to learn. The curly-brace syntax of C# will be instantly recognizable to anyone familiar with C, C++ or Java. Developers who know any of these languages are typically able to begin to work productively in C# within a very short time. C# syntax simplifies many of the complexities of C++ and provides powerful features such as nullable value types, enumerations, delegates, lambda expressions, and direct memory access, which are not found in Java. C# supports generic methods and types, which provide increased type safety and performance, and iterators, which enable implementers of collection classes to define custom iteration behaviors that are simple to use by client code. Language-Integrated Query (LINQ) expressions make the strongly-typed query a first-class language construct.
```
Comment (Youssef1313) [inline docs/csharp/getting-started/introduction-to-the-csharp-language-and-the-net-framework.md]: Enumerations does exists in Java AFAIK.
For a reference, check [Java Enums - W3Schools](https://www.w3schools.com/java/java_enums.asp).

Lambda expressions were added to Java 8.
For reference, check [Lambda Expressions in Java 8 - GeeksforGeeks](https://www.geeksforgeeks.org/lambda-expressions-java-8/)

It's not good to refer to C# language features as non-existent in Java while they do.
This suggestions removes the features I found existed in Java. (I didn't look if other features being mentioned exists in Java.)

Another suggestions, is to just remove "which are not found in Java" and keep the Lambda expressions and Enumerations.

```suggestion
 C# syntax is highly expressive, yet it is also simple and easy to learn. The curly-brace syntax of C# will be instantly recognizable to anyone familiar with C, C++, or Java. Developers who know any of these languages are typically able to begin to work productively in C# within a very short time. C# syntax simplifies many of the complexities of C++ and provides powerful features such as nullable value types, delegates, and direct memory access, which are not found in Java. C# supports generic methods and types, which provide increased type safety and performance, and iterators, which enable implementers of collection classes to define custom iteration behaviors that are simple to use by client code. Language-Integrated Query (LINQ) expressions make the strongly-typed query a first-class language construct.  
```

File: docs/csharp/getting-started/introduction-to-the-csharp-language-and-the-net-framework.md [hunk 3]
```diff
@@ -50,7 +50,7 @@ C# is an elegant and type-safe object-oriented language that enables developers
...
+ When the C# program is executed, the assembly is loaded into the CLR, which might take various actions based on the information in the manifest. Then, if the security requirements are met, the CLR performs just in time (JIT) compilation to convert the IL code to native machine instructions. The CLR also provides other services related to automatic garbage collection, exception handling, and resource management. Code that's executed by the CLR is sometimes referred to as "managed code", in contrast to "unmanaged code", which is compiled into native machine language that targets a specific system. The following diagram illustrates the compile-time and run-time relationships of C# source code files, the .NET Framework class libraries, assemblies, and the CLR.
```
Comment (Youssef1313) [inline docs/csharp/getting-started/introduction-to-the-csharp-language-and-the-net-framework.md]:
```suggestion
 When the C# program is executed, the assembly is loaded into the CLR, which might take various actions based on the information in the manifest. Then, if the security requirements are met, the CLR performs Just-In-Time (JIT) compilation to convert the IL code to native machine instructions. The CLR also provides other services related to automatic garbage collection, exception handling, and resource management. Code that's executed by the CLR is sometimes referred to as "managed code", in contrast to "unmanaged code", which is compiled into native machine language that targets a specific system. The following diagram illustrates the compile-time and run-time relationships of C# source code files, the .NET Framework class libraries, assemblies, and the CLR.  
```

File: docs/csharp/getting-started/introduction-to-the-csharp-language-and-the-net-framework.md [hunk 4]
```diff
@@ -50,7 +50,7 @@ C# is an elegant and type-safe object-oriented language that enables developers
...
+ When the C# program is executed, the assembly is loaded into the CLR, which might take various actions based on the information in the manifest. Then, if the security requirements are met, the CLR performs just in time (JIT) compilation to convert the IL code to native machine instructions. The CLR also provides other services related to automatic garbage collection, exception handling, and resource management. Code that's executed by the CLR is sometimes referred to as "managed code", in contrast to "unmanaged code", which is compiled into native machine language that targets a specific system. The following diagram illustrates the compile-time and run-time relationships of C# source code files, the .NET Framework class libraries, assemblies, and the CLR.  
   
  ![From C&#35; source code to machine execution](./media/introduction-to-the-csharp-language-and-the-net-framework/net-architecture-relationships.png)
```
Comment (Youssef1313) [inline docs/csharp/getting-started/introduction-to-the-csharp-language-and-the-net-framework.md]: Not sure why the HTML entity was used. Does the "#" cause any problems ?

```suggestion
 ![From C# source code to machine execution](./media/introduction-to-the-csharp-language-and-the-net-framework/net-architecture-relationships.png)  
```

File: docs/csharp/programming-guide/concepts/linq/how-to-stream-xml-fragments-with-access-to-header-information.md [hunk 5]
```diff
@@ -10,7 +10,7 @@ One option is to write your application using <xref:System.Xml.XmlReader>. Howev
...
+ Streaming techniques are best applied in situations where you need to process the source document only once, and you can process the elements in document order. Certain standard query operators, such as <xref:System.Linq.Enumerable.OrderBy%2A>, iterate their source, collect all of the data, sort it, and then finally yield the first item in the sequence. If you use a query operator that materializes its source before yielding the first item, you will not retain a small memory footprint.  
   
 ## Example  
  Sometimes the problem gets just a little more interesting. In the following XML document, the consumer of your custom axis method also has to know the name of the customer that each item belongs to.
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/concepts/linq/how-to-stream-xml-fragments-with-access-to-header-information.md]:
```suggestion

Sometimes the problem gets just a little more interesting. In the following XML document, the consumer of your custom axis method also has to know the name of the customer that each item belongs to.  
```

File: docs/csharp/programming-guide/concepts/linq/linq-and-file-directories.md [hunk 6]
```diff
@@ -3,32 +3,33 @@ title: "LINQ and File Directories (C#)"
 ms.date: 07/20/2015
 ms.assetid: b66c55e4-0f72-44e5-b086-519f9962335c
 ---
-# LINQ and File Directories (C#)
-Many file system operations are essentially queries and are therefore well-suited to the LINQ approach.  
+# LINQ and file directories (C#)
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/concepts/linq/linq-and-file-directories.md]: Same change in title metadata ?

**Review by Youssef1313 — CHANGES_REQUESTED** [review]

File: docs/csharp/getting-started/introduction-to-the-csharp-language-and-the-net-framework.md [hunk 7]
```diff
@@ -7,16 +7,16 @@ helpviewer_keywords:
...
+ C# syntax is highly expressive, yet it is also simple and easy to learn. The curly-brace syntax of C# will be instantly recognizable to anyone familiar with C, C++ or Java. Developers who know any of these languages are typically able to begin to work productively in C# within a very short time. C# syntax simplifies many of the complexities of C++ and provides powerful features such as nullable value types, enumerations, delegates, lambda expressions, and direct memory access, which are not found in Java. C# supports generic methods and types, which provide increased type safety and performance, and iterators, which enable implementers of collection classes to define custom iteration behaviors that are simple to use by client code. Language-Integrated Query (LINQ) expressions make the strongly-typed query a first-class language construct.
```
Comment (gewarren) [inline docs/csharp/getting-started/introduction-to-the-csharp-language-and-the-net-framework.md]:
```suggestion
C# syntax is highly expressive, yet it is also simple and easy to learn. The curly-brace syntax of C# will be instantly recognizable to anyone familiar with C, C++, or Java. Developers who know any of these languages are typically able to begin to work productively in C# within a very short time. C# syntax simplifies many of the complexities of C++ and provides powerful features such as nullable value types, enumerations, delegates, lambda expressions, and direct memory access. C# supports generic methods and types, which provide increased type safety and performance, and iterators, which enable implementers of collection classes to define custom iteration behaviors that are simple to use by client code. Language-Integrated Query (LINQ) expressions make the strongly typed query a first-class language construct.  
```

File: docs/csharp/getting-started/introduction-to-the-csharp-language-and-the-net-framework.md (hunk 2 above)
Comment (gewarren) [reply docs/csharp/getting-started/introduction-to-the-csharp-language-and-the-net-framework.md]: Thanks for noticing this. I just removed "which are not found in Java".

**Review by BillWagner — APPROVED** [review]
Everything LGTM @gewarren 

You can :shipit: when ready.

**Comment by gewarren** [comment]
Thank you Bill, and happy new year!

**Pull request merged**